quinn/src/streams.rs

```diff
@@ -149,6 +151,39 @@ where
         conn.wake();
     }
 
+    /// Completes if/when the peer stops the stream, yielding the error code
+    pub fn stopped(&mut self) -> Stopped<'_, S> {
+        Stopped { stream: self }
+    }
+
+    #[doc(hidden)]
+    pub fn poll_stopped(&mut self, cx: &mut Context) -> Poll<Result<VarInt, WriteError>> {
```

Comment:
It feels a little odd to return `WriteError` here due to the `WriteError::Stopped` case; a user might reasonably be confused about how a stopped stream is represented. Maybe it's worth a dedicated error type? Another option might be to return `WriteError` directly--after all, it does have exactly the cases we care about--but that leaves the user without an ergonomic way to distinguish errors from success, which doesn't seem great. Open to discussion.

Reply:
I agree with you but I chose not to add another type for this (at the moment). I think returning "XXXError" type not in an `Err` variant is weird, unless the function call is explicit about error management. 

I guess I prefer the custom type (`WriteError`, but without `Stopped` variant).